File: app/test_db_chats.mjs
import sqlite3 from 'sqlite3';
import { open } from 'sqlite';
import path from 'path';

async function queryDb(dbPath, query) {
    try {
        const db = await open({ filename: dbPath, driver: sqlite3.Database });
        const rows = await db.all(query);
        await db.close();
        return rows;
    } catch (e) {
        return { error: e.message };
    }
}

async function main() {
    const dbPaths = [
        path.join(process.cwd(), 'satis_db', 'ziyaretler.db'),
        path.join(process.cwd(), 'satis_db', 'customer_chats.db'),
        path.join(process.cwd(), 'satis_db', 'internal_chats.db'),
        path.join(process.cwd(), 'satis_db', 'quota_hub.db')
    ];

    console.log("SQLITE DATABASE QUERY SEARCH:");
    
    for (const dbp of dbPaths) {
        console.log(`\n--- DB: ${path.basename(dbp)} ---`);
        // Tabloları listele
        const tables = await queryDb(dbp, "SELECT name FROM sqlite_master WHERE type='table'");
        console.log("Tables:", JSON.stringify(tables));
        
        if (Array.isArray(tables)) {
            for (const t of tables) {
                // Her tabloda telegram veya chat ile ilgili kolonları tara
                const info = await queryDb(dbp, `PRAGMA table_info(${t.name})`);
                const cols = info.map(c => c.name);
                
                // Ufuk veya telegram araması yap
                if (cols.length > 0) {
                    const searchQueries = [
                        `SELECT * FROM ${t.name} WHERE ${cols.map(c => `${c} LIKE '%ufuk%'`).join(' OR ')} LIMIT 10`,
                        `SELECT * FROM ${t.name} WHERE ${cols.map(c => `${c} LIKE '%1190118497%'`).join(' OR ')} LIMIT 10`
                    ];
                    
                    for (const sq of searchQueries) {
                        const results = await queryDb(dbp, sq);
                        if (Array.isArray(results) && results.length > 0) {
                            console.log(`Matches in ${t.name}:`, JSON.stringify(results, null, 2));
                        }
                    }
                }
            }
        }
    }
}

main();

File: app/test_db_chats.py
import sqlite3
import os
import json

db_paths = [
    os.path.join(os.getcwd(), 'satis_db', 'ziyaretler.db'),
    os.path.join(os.getcwd(), 'satis_db', 'customer_chats.db'),
    os.path.join(os.getcwd(), 'satis_db', 'internal_chats.db'),
    os.path.join(os.getcwd(), 'satis_db', 'quota_hub.db')
]

print("PYTHON SQLITE SEARCH:")

for dbp in db_paths:
    if not os.path.exists(dbp):
        print(f"File not found: {dbp}")
        continue
    
    print(f"\n--- DB: {os.path.basename(dbp)} ---")
    try:
        conn = sqlite3.connect(dbp)
        cursor = conn.cursor()
        
        # Tabloları listele
        cursor.execute("SELECT name FROM sqlite_master WHERE type='table'")
        tables = [t[0] for t in cursor.fetchall()]
        print(f"Tables: {tables}")
        
        for table in tables:
            cursor.execute(f"PRAGMA table_info({table})")
            cols = [c[1] for c in cursor.fetchall()]
            
            # Arama sorguları
            for search_term in ['ufuk', '1190118497', 'telegram']:
                where_clause = " OR ".join([f"{col} LIKE '%{search_term}%'" for col in cols])
                if not where_clause:
                    continue
                
                query = f"SELECT * FROM {table} WHERE {where_clause} LIMIT 10"
                try:
                    cursor.execute(query)
                    rows = cursor.fetchall()
                    if rows:
                        print(f"Matches in table {table} for '{search_term}':")
                        for row in rows:
                            print(row)
                except Exception as table_err:
                    pass
                    
        conn.close()
    except Exception as e:
        print(f"Error reading DB: {e}")
